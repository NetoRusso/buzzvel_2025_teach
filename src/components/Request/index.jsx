
import styles from "./styles.module.css";

const Request = () => {

  return (
    <section className={styles.container} aria-label="Request">
      <div className={styles.content}>
        <h2 className={styles.title}>
          Ready for your next project?
        </h2>
        <p className={styles.paragraph}>
          Sit elit feugiat turpis sed integer integer accumsan turpis.
        </p>
        <form className={styles.form}>
          <label htmlFor="email" className={styles.label}>Email</label>
          <input type="email" id="email" name="email" placeholder="Enter your email" required className={styles.input} />
          <label htmlFor="message" className={styles.label}>Message</label>
          <textarea id="message" name="message" placeholder="What are you say ?" required className={styles.textarea} />
          <button type="submit" className={styles.btn}>
            Request Demo
          </button>
        </form>
      </div>
    </section>
  )

};

export default Request;
