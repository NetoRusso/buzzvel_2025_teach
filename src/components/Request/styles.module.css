.container {
  width: 100%;
  height: auto;
  min-height: 679px;
  background-color: var(--orange-600);
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 16px;
}

.content {
  width: 100%;
  max-width: 1280px;
  height: auto;
  min-height: 514px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 32px;
}

.title {
  font-size: clamp(2rem, 1.437rem + 2.2923vw, 3.5rem);
  line-height: 110%;
  font-weight: 800;
  color: var(--background-white);
  text-align: center;
}

.paragraph {
  font-size: clamp(1.125rem, 0.7966rem + 1.3372vw, 2rem);
  line-height: 160%;
  font-weight: 400;
  color: var(--background-white);
  text-align: center;
}



.form {
  width: 100%;
  max-width: 570px;
  height: auto;
  padding: 16px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 16px;
}

.label {
  font-size: 18px;
  font-weight: 500;
  color: var(--background-white);
  align-self: start;
  padding-left: 8px;
  margin-bottom: -4px;
}

.input {
  width: 100%;
  height: 50px;
  border-radius: 10px;
  background-color: var(--background-white);
  padding: 8px 16px;
  border: none;
  color: var(--blue-Gray-600);
  font-weight: 500;
  font-size: 14px;
  font-family: 'inter';
}

.input:focus {
  border: 5px solid var(--blue-600);
  outline: none;
}

.input::placeholder {
  color: var(--blue-Gray-600);
  font-weight: 500;
  font-size: 14px;
  font-family: "inter";
}

.textarea {
  width: 100%;
  height: auto;
  min-height: 80px;
  border: none;
  border-radius: 10px;
  padding: 16px;
  color: var(--blue-Gray-600);
  font-weight: 500;
  font-size: 14px;
  font-family: 'inter';
}

.textarea:focus {
  border: 5px solid var(--blue-600);
  outline: none;
}

.textarea::placeholder {
  color: var(--blue-Gray-600);
  font-weight: 500;
  font-size: 14px;
  font-family: 'inter';
}

.btn {
  width: 100%;
  max-width: 260px;
  min-width: 201px;
  height: clamp(3.5rem, 3.3123rem + 0.7641vw, 4rem);
  max-height: 64px;
  min-height: 56px;
  border-radius: 8px;
  border: none;
  background-color: var(--blue-Gray-900);
  font-size: clamp(1rem, 0.8123rem + 0.7641vw, 1.5rem);
  font-weight: 700;
  color: var(--background-white);
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  margin-top: 24px;
}

.btn:hover {
  /* scale: 0.95; */
  background-color: var(--blue-600);
}