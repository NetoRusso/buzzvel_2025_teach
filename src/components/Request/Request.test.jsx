import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Request from './index';

describe('Request Component', () => {
  test('renders static text content correctly', () => {
    render(<Request />);
    expect(screen.getByRole('heading', { name: /Ready for your next project\?/i })).toBeInTheDocument();
    expect(screen.getByText(/Sit elit feugiat turpis sed integer integer accumsan turpis\./i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Email/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Message/i)).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Request Demo/i })).toBeInTheDocument();
  });

  test('renders form fields with correct attributes', () => {
    render(<Request />);

    const emailInput = screen.getByLabelText(/Email/i);
    expect(emailInput).toBeInTheDocument();
    expect(emailInput).toHaveAttribute('type', 'email');
    expect(emailInput).toHaveAttribute('id', 'email');
    expect(emailInput).toHaveAttribute('name', 'email');
    expect(emailInput).toHaveAttribute('placeholder', 'Enter your email');
    expect(emailInput).toBeRequired();

    const messageTextarea = screen.getByLabelText(/Message/i);
    expect(messageTextarea).toBeInTheDocument();
    expect(messageTextarea).toHaveAttribute('id', 'message');
    expect(messageTextarea).toHaveAttribute('name', 'message');
    expect(messageTextarea).toHaveAttribute('placeholder', 'What are you say ?');
    expect(messageTextarea).toBeRequired();

    const submitButton = screen.getByRole('button', { name: /Request Demo/i });
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toHaveAttribute('type', 'submit');
  });

  test('has correct aria-label for accessibility on the section', () => {
    render(<Request />);
    expect(screen.getByRole('region', { name: 'Request' })).toBeInTheDocument();
  });

  test('allows typing in form fields', () => {
    render(<Request />);

    const emailInput = screen.getByLabelText(/Email/i);
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    expect(emailInput).toHaveValue('test@example.com');

    const messageTextarea = screen.getByLabelText(/Message/i);
    fireEvent.change(messageTextarea, { target: { value: 'Hello there!' } });
    expect(messageTextarea).toHaveValue('Hello there!');
  });

  test('matches snapshot', () => {
    const { container } = render(<Request />);
    expect(container.firstChild).toMatchSnapshot();
  });
});