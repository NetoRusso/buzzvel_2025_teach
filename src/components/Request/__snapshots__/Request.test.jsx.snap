// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Component matches snapshot 1`] = `
<section
  aria-label="Request"
  class="container"
>
  <div
    class="content"
  >
    <h2
      class="title"
    >
      Ready for your next project?
    </h2>
    <p
      class="paragraph"
    >
      Sit elit feugiat turpis sed integer integer accumsan turpis.
    </p>
    <form
      class="form"
    >
      <label
        class="label"
        for="email"
      >
        Email
      </label>
      <input
        class="input"
        id="email"
        name="email"
        placeholder="Enter your email"
        required=""
        type="email"
      />
      <label
        class="label"
        for="message"
      >
        Message
      </label>
      <textarea
        class="textarea"
        id="message"
        name="message"
        placeholder="What are you say ?"
        required=""
      />
      <button
        class="btn"
        type="submit"
      >
        Request Demo
      </button>
    </form>
  </div>
</section>
`;
